packages/cli/src/commands/init-client.ts

```diff
@@ -0,0 +1,108 @@
+import type * as yargs from 'yargs';
+import { getSchemaFolder } from '../utils';
+import {
+  logDetail,
+  logError,
+  logInfo,
+  logUnknownError,
+  logVerbose
+} from '@syftdata/common/lib/utils';
+import * as fs from 'fs';
+import * as path from "path";
+import { handler as generateFromDir } from './generate';
+import { fetchRemoteData } from '../init/destination';
+import type { AST } from '@syftdata/common/lib/types';
+import { getEventShemas } from '../init/local';
+import { writeTestSpecs } from '../publish/remote';
+import { serialize } from '@syftdata/codehandler';
+import { ModuleKind, Project, ScriptTarget } from 'ts-morph';
+import {getClientPackage} from "../config/pkg";
+import {PluginDependency} from "@syftdata/client/src";
+import {generate} from "../codegen/generators/ts_generator";
+
+export interface Params {
+  platform: string;
+  product: string;
+  outDir: string;
+  apikey?: string;
+  branch?: string;
+  remote: string;
+  force: boolean;
+}
+
+export const command = 'init-client';
+export const desc = 'Initialize Syft project. Generates Event models.';
+export const builder = (y: yargs.Argv): yargs.Argv => {
+  return y;
+};
+
+function getMetricsProvider(packageJson?: any): string | null {
+  if (packageJson != null) {
+    const dependencies = packageJson.dependencies;
+
+    for (const value in PluginDependency) {
+      if (dependencies.hasOwnProprety(value) === true) {
+        return PluginDependency[value];
+      }
+    }
+  }
+  // nothing found
+  return null;
+}
+
+async function createSyftTS(ast: AST, folder: string): Promise<void> {
+  const metricsProvider = await getClientPackage().then((val) => {
+    return getMetricsProvider(val.json);
+  });
+  if (metricsProvider == null) {
+    logError(`:warning: No package.json file found. Exiting...`);
+    return;
+  }
+  const src = getSchemaFolder();
+  logVerbose(`Writing syft.ts to src directory...`);
+
+  const srcPath = path.relative('../../assets', 'syft.ts');
+  generate()
+
+}
+
+export async function handler(): Promise<void> {
```

Comment:
this looks like a clone of `init.ts` lets just update init ? 

- you could put an optional flag to disable `syft.ts` generation if you want.

Reply:
checked in by accident. this is a WIP. added label